src/logic/import/voc/VOCImporter.ts

```diff
@@ -0,0 +1,105 @@
+                }
+
+                onSuccess(
+                    Array.from(Object.values(inputImagesData)),
+                    Array.from(Object.values(results.labelNames))
+                );
+            }).catch((error: Error) => onFailure(error));
+        } catch (error) {
+            onFailure(error as Error)
+        }
+    }
+
+    private loadAndParseFiles(files: File[]): Promise<VOCImportResult> {
+        const parser = new DOMParser();
+
+        return Promise.all(files.map(file => file.text())).then(textFiles => 
+            textFiles.reduce((current, fileData) => 
+            VOCImporter.parseDocumentIntoImageData(parser.parseFromString(fileData, 'application/xml'), current), 
+                {
+                    labelNames: {},
+                    fileParseResults: [],
+                } as VOCImportResult)
+            );
+    }
+
+    protected static parseDocumentIntoImageData(document: Document, { fileParseResults, labelNames }: VOCImportResult): VOCImportResult {
+        const root = document.getElementsByTagName('annotation')[0];
+        const filename = root.getElementsByTagName('filename')[0].textContent;
+        
+        const [labeledBoxes, newLabelNames] = this.parseAnnotationsFromFileString(document, labelNames);
+
+        return {
+            labelNames: newLabelNames,
+            fileParseResults: fileParseResults.concat({
+                filename,
+                labeledBoxes
+            }),
+        };
+    }
+
+    protected static parseAnnotationsFromFileString(document: Document, labelNames: Record<string, LabelName>): 
+        [LabelRect[], Record<string, LabelName>] {
+        const newLabelNames: Record<string, LabelName> = Object.assign(labelNames);
+        return [Array.from(document.getElementsByTagName('object')).map(d => {
+            const labelName = d.getElementsByTagName('name')[0].textContent;
+            const bbox = d.getElementsByTagName('bndbox')[0];
+            const xmin = parseInt(bbox.getElementsByTagName('xmin')[0].textContent);
+            const xmax = parseInt(bbox.getElementsByTagName('xmax')[0].textContent);
+            const ymin = parseInt(bbox.getElementsByTagName('ymin')[0].textContent);
+            const ymax = parseInt(bbox.getElementsByTagName('ymax')[0].textContent);
+            const rect = {
+                x: xmin,
+                y: ymin,
+                height: ymax - ymin,
+                width: xmax - xmin, 
+            };
+            
+            if (!newLabelNames[labelName]) {
+                newLabelNames[labelName] = LabelUtil.createLabelName(labelName);
+            }
+            
+            const labelId = newLabelNames[labelName].id;
+
+            return LabelUtil.createLabelRect(labelId, rect);
+        }), newLabelNames];
+    }
+
+    private static mapImageData(): Record<string, ImageData> {
+        return LabelsSelector.getImagesData().reduce(
+            (c: Record<string, ImageData>, i: ImageData) => {
```

Comment:
I didn't notice that before (really sorry for that) but I tend to not use one-character variable names. So if that is okay maybe we could rename those to something more meaningful. :)